### PR TITLE
[IOTDB-3114] NodeInfo snapshot interface

### PR DESCRIPTION
**Pull request by cigarl**

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java [hunk 1]
```diff
@@ -272,15 +278,129 @@ public int generateNextDataNodeId() {
...
+  // as drainingDataNodes is not currently implemented, manually set it to validate the test
+  @TestOnly
+  public void setDrainingDataNodes(Set<TDataNodeLocation> tDataNodeLocations) {
+    drainingDataNodes.addAll(tDataNodeLocations);
   }
 
-  public void deserialize(ByteBuffer buffer) {
-    // TODO: Deserialize DataNodeInfo
+  @TestOnly
+  public int getNextDataNodeId() {
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java:395]: This interface is not TestOnly

**Review by CRZbulabula — CHANGES_REQUESTED** [review]
PTAL~

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java (hunk 1 above)
Comment (cigarl) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java]: fixed

**Review by CRZbulabula — APPROVED** [review]
LGTM.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**